apps/studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx

```diff
@@ -132,28 +169,70 @@ const GridHeaderActions = ({ table }: GridHeaderActionsProps) => {
-              table.rls_enabled ? (
-                <IconLock strokeWidth={2} size={14} />
+        <div>
+          {(table.rls_enabled || showRLSWarning) && (
+            <>
+              {table.rls_enabled ? (
+                <div className="flex items-center gap-1">
+                  <Link passHref href={`/project/${projectRef}/auth/policies?search=${table.id}`}>
+                    <Button
+                      type={table.rls_enabled ? 'default' : 'warning'}
+                      className="group"
+                      icon={
+                        policies.length > 0 ? (
+                          <span>
+                            <span className="text-right text-xs rounded-xl px-2 py-0.5 bg-surface-200 dark:bg-surface-100 text-brand-1100">
+                              {policies.length}
+                            </span>
+                          </span>
+                        ) : (
+                          <IconPlusCircle size={12} />
+                        )
+                      }
+                    >
+                      <span className="flex-1 text-left">
+                        Auth {policies.length > 1 ? 'policies' : 'policy'}
+                      </span>
+                    </Button>
+                  </Link>
+                </div>
               ) : (
-                <IconAlertCircle strokeWidth={2} size={14} />
-              )
-            }
-          >
-            <Link href={`/project/${projectRef}/auth/policies?search=${table.id}`}>
-              {!table.rls_enabled
-                ? 'RLS is not enabled'
-                : `${policies.length == 0 ? 'No' : policies.length} active RLS polic${
-                    policies.length > 1 || policies.length == 0 ? 'ies' : 'y'
-                  }`}
-            </Link>
-          </Button>
-        )}
+                <Popover_Shadcn_ open={open} onOpenChange={() => setOpen(!open)} modal={false}>
+                  <PopoverTrigger_Shadcn_ asChild>
+                    <Button type="warning" icon={<IconLock size={14} />}>
+                      RLS Disabled
+                    </Button>
+                  </PopoverTrigger_Shadcn_>
+                  <PopoverContent_Shadcn_ className="w-48 min-w-[395px]">
+                    <div className="grid gap-2 text-sm px-2 py-4">
+                      <h3 className="flex items-center gap-2">
+                        <IconLock size={14} /> Row Level Security (RLS)
+                      </h3>
+                      <div className="grid gap-2 mt-1">
+                        <p>
+                          You can restrict and control who can read, write and update data in this
+                          table using Row Level Security.
+                        </p>
+                        <p>
+                          With RLS enabled, anonymous users will not be able to read/write data in
+                          the table.
+                        </p>
+                        <p className="mt-2">
```

Comment:
Button doesn't need to be in a p

Reply:
Might be because the button stretches.

In which case, change to use flex rather than grid tailwind.

Or use div rather than p, makes more semantic sense